api/proto/message/message.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+package message;
+
+enum MessageType {
+  UNKNOWN = 0;
+  NEWNODE_BOOTNODE = 1;
+  BOOTNODE_NEWNODE = 2;
+  NEWNODE_BEACON = 3;
+  BEACON_NEWNODE = 4;
+}
+
+// This is universal message for all communication protocols.
+// There are different Requests for different message types.
+// As we introduce a new type of message just add a new MessageType and new type of request in Message.
+//
+// The request field will be either one of the structure corresponding to the MessageType type.
+message Message {
```

Comment:
Not sure if it's good to have a single wraper for all messages. Maybe it's too early to decide that. But migrating to grpc is good.

Reply:
I don't know if we need to migrate to grpc for all messages.  @rlan35 , you mean using protobuf?

Reply:
yea, protobuf.

Reply:
I will use this for beacon chain first.

Reply:
grpc is actually just one feature of protobuf